esphome/components/online_image/bmp_image.cpp

```diff
@@ -0,0 +1,72 @@
+#include "bmp_image.h"
+
+#ifdef USE_ONLINE_IMAGE_BMP_SUPPORT
+
+#include "esphome/components/display/display.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace online_image {
+
+static const char *const TAG = "online_image.bmp";
+
+int HOT BmpDecoder::decode(uint8_t *buffer, size_t size) {
+  size_t index = 0;
+  this->decoded_bytes_ += size;
+  if (this->current_index_ == 0 && this->decoded_bytes_ > 14) {
+    // Check if the file is a BMP image
+    if (buffer[0] != 'B' || buffer[1] != 'M') {
+      ESP_LOGE(TAG, "Not a BMP file");
+      return 0;
+    }
+
+    this->total_size_ = encode_uint32(buffer[5], buffer[4], buffer[3], buffer[2]);
+    this->data_offset_ = encode_uint32(buffer[13], buffer[12], buffer[11], buffer[10]);
+
+    this->current_index_ = 14;
+    index = 14;
+  }
+  if (this->current_index_ == 14 && this->decoded_bytes_ > data_offset_) {
+    this->width_ = encode_uint32(buffer[21], buffer[20], buffer[19], buffer[18]);
+    this->height_ = encode_uint32(buffer[25], buffer[24], buffer[23], buffer[22]);
+    this->bits_per_pixel_ = encode_uint16(buffer[29], buffer[28]);
+    this->compression_method_ = encode_uint32(buffer[33], buffer[32], buffer[31], buffer[30]);
+    this->image_data_size_ = encode_uint32(buffer[37], buffer[36], buffer[35], buffer[34]);
+    this->color_table_entries_ = encode_uint32(buffer[49], buffer[48], buffer[47], buffer[46]);
+
+    switch (this->bits_per_pixel_) {
+      case 1:
+        this->width_bytes_ = (this->width_ % 8 == 0) ? (this->width_ / 8) : (this->width_ / 8 + 1);
+        break;
+      default:
+        ESP_LOGE(TAG, "Unsupported bits per pixel: %d", this->bits_per_pixel_);
+        return 0;
```

Comment:
```suggestion
        return -2;
```
Maybe we should define some constants for possible errors in the "ImageDecoder" class

Reply:
I like this idea